stripe_official.php

```diff
@@ -1605,11 +1609,14 @@ public function hookHeader()
             return;
         }
 
-        $currency = $this->context->currency->iso_code;
-        $address = new Address($this->context->cart->id_address_invoice);
-        $amount = $this->context->cart->getOrderTotal();
+        $cart = $this->context->cart;
+
+        $address = new Address($cart->id_address_invoice);
+        $currency = new Currency($cart->id_currency);
+        $amount = $cart->getOrderTotal();
         $amount = Tools::ps_round($amount, 2);
-        $amount = $this->isZeroDecimalCurrency($currency) ? $amount : $amount * 100;
+        $amount = $this->isZeroDecimalCurrency($currency->iso_code) ? $amount : $amount * 100;
+        $amount = Tools::ps_round($amount);
```

Comment:
Becarefull when you arrond Japan currency have 4 digits as decimal and specific behaviour.